var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -97,6 +98,10 @@ class Hdf5(AutotoolsPackage):
     patch('h5f90global-mult-obj-same-equivalence-same-common-block.patch',
           when='@1.10.1%intel@18')
 
+    # Turn line comments into block comments to conform with pre C99 language
+    # standards
+    patch('pre-c99-comments.patch', when='@1.8.10')
```

Comment:
@mmpozulp I'm curious if you wrote this patch yourself or if it comes from somewhere. In the latter case could you include the link? Also I'm curious if `hdf5` would build successfully depending on the compiler version. Do versions of `hdf5` after `1.8.10` conform to C99 language standards? If you're not sure, IMO you don't have to find out but I think it would be worth mentionig that here.

Reply:
I wrote the patch myself. I'm not 100% sure, but I think the hdf5 1.8.10 build system is setting a compiler flag to use the ISO C90 language standard, which is why my compiler errors-out when it sees the single-line comment syntax. Versions of hdf5 after 1.8.10 must either conform to pre-C99 or neglect to ask for pre-C99 language standards from their compiler. 

If the hdf5 1.8.10 build system is indeed setting a compiler flag to ask for a pre-C99 language standard, an alternative to patching the non-conformant src would be to patch the build system to prevent the pre-C99 from being added.

Reply:
@scheibelp The cflag that hdf5 adds to ask for pre-C99 is -ansi. Here's the relevant info from spack-build.out
```text
SUMMARY OF THE HDF5 CONFIGURATION
=================================
...

Compiling Options:
------------------
...
H5_CFLAGS:   -ansi -pedantic -Wall -W -Wundef -Wshadow -Wpointer-arith -Wbad-function-cast -Wcast-qual -Wcast-align -Wwrite-strings -Wconversion -Waggregate-return -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls -Wnested-externs -Winline -O3 -fomit-frame-pointer -finline-functions
...
```
I didn't ask for '-ansi', hdf5 added it on its own.

Reply:
@scheibelp Thus, any compiler supporting '-ansi' should fail to compile hdf5. I confirmed that both of the following fail
```console
$ spack install hdf5@1.8.10 %gcc@4.9.3
$ spack install hdf5@1.8.10 %intel@17.0.0
``` 
so presumably this effects many compilers. I'm on llnl's toss3 machine rzgenie (intel x86-64).

Reply:
@mmpozulp Thank you for the info. I think the following is worth stating a summary of the above in a comment, e.g.: 

> Versions of `hdf5` after `1.8.10` don't require this patch, either because they conform to pre-C99 or neglect to ask for pre-C99 language standards from their compiler

Regarding 

> If the hdf5 1.8.10 build system is indeed setting a compiler flag to ask for a pre-C99 language standard, an alternative to patching the non-conformant src would be to patch the build system to prevent the pre-C99 flag from being added.

IMO that would be preferable but I definitely wouldn't require it here; IMO it's preferable because it would be easier to detect this as a pattern (if this issue came up with other packages) and absorb it into a parent class if it were expressed as a build option vs. as a patch.

Reply:
@scheibelp Fine with me to merge as soon as you think it's fine.